podpac/core/coordinates/test/test_array_coordinates1d.py

```diff
@@ -742,41 +486,66 @@ def test_index(self):
+        c = ArrayCoordinates1d(values)
 
-        c2 = c[1:3]
-        assert_equal(c2.segment_lengths, [0.2, 0.3])
+        # point
+        area_bounds = c.get_area_bounds(None)
+        assert_equal(area_bounds, [0.0, 6.0])
 
-        c2 = c[[2, 1]]
-        assert_equal(c2.segment_lengths, [0.3, 0.2])
+        # uniform
+        area_bounds = c.get_area_bounds(0.5)
+        assert_equal(area_bounds, [-0.5, 6.5])
 
-        c2 = c[[]]
-        assert_equal(c2.segment_lengths, [])
+        # segment
+        area_bounds = c.get_area_bounds([-0.2, 0.7])
+        assert_equal(area_bounds, [-0.2, 6.7])
 
-        # uniform segment_lengths
-        c = ArrayCoordinates1d([1, 2, 4, 5], segment_lengths=0.5)
+        # polygon (i.e. there would be corresponding offets for another dimension)
+        area_bounds = c.get_area_bounds([-0.2, -0.5, 0.7, 0.5])
+        assert_equal(area_bounds, [-0.5, 6.7])
+
+        # boundaries
+        area_bounds = c.get_area_bounds([[-0.4, 0.1], [-0.3, 0.2], [-0.2, 0.3], [-0.1, 0.4]])
+        assert_equal(area_bounds, [-0.4, 6.4])
+
+    def test_get_area_bounds_datetime(self):
+        values = make_coord_array(["2017-01-02", "2017-01-01", "2019-01-01", "2018-01-01"])
+        c = ArrayCoordinates1d(values)
 
-        c2 = c[1]
-        assert c2.segment_lengths == 0.5
+        # point
+        area_bounds = c.get_area_bounds(None)
+        assert_equal(area_bounds, make_coord_array(["2017-01-01", "2019-01-01"]))
 
-        c2 = c[1:3]
-        assert c2.segment_lengths == 0.5
+        # uniform
+        area_bounds = c.get_area_bounds("1,D")
+        assert_equal(area_bounds, make_coord_array(["2016-12-31", "2019-01-02"]))
 
-        c2 = c[[2, 1]]
-        assert c2.segment_lengths == 0.5
+        area_bounds = c.get_area_bounds("1,M")
+        assert_equal(area_bounds, make_coord_array(["2016-12-01", "2019-02-01"]))
 
-        c2 = c[[]]
-        assert c2.segment_lengths == 0.5
+        area_bounds = c.get_area_bounds("1,Y")
+        assert_equal(area_bounds, make_coord_array(["2016-01-01", "2020-01-01"]))
+
+        # segment
+        area_bounds = c.get_area_bounds(["0,h", "12,h"])
+        assert_equal(area_bounds, make_coord_array(["2017-01-01 00:00", "2019-01-01 12:00"]))
+
+    def test_get_area_bounds_empty(self):
+        c = ArrayCoordinates1d([])
+        area_bounds = c.get_area_bounds(1.0)
+        assert np.all(np.isnan(area_bounds))
+
+    @pytest.mark.xfail(reason="spec uncertain")
+    def test_get_area_bounds_overlapping(self):
+        values = np.array([0.0, 1.0, 4.0, 6.0])
+        c = ArrayCoordinates1d(values)
 
-        # inferred segment_lengths
-        c = ArrayCoordinates1d([1, 2, 4, 7], ctype="left")
-        c2 = c[1]
-        assert c2.segment_lengths == 2.0 or np.array_equal(c2.segment_lengths, [2.0])
+        area_bounds = c.get_area_bounds([[-0.1, 0.1], [-10.0, 10.0], [-0.1, 0.1], [-0.1, 0.1]])
+        assert_equal(area_bounds, [-11.0, 11.0])
```

Comment:
@mpu-creare And the tests for get_area_bounds, showing `boundary` in use. Again, I think this is pretty clean.

Reply:
That's pretty good. I'm impressed you covered the overlapping case.